### PR TITLE
Add a 'sample' keyword to apply time interval between SDO data from JSOC

### PR DESCRIPTION
**Pull request by jongyeob**
From #1421 

The hash of last commit was changed after a cherry-picking, so #1421 was automatically closed.

This commit reflects suggestions of @Cadair. I didn't recognize the Sample class in VSO.

Now the sample class has a input as only seconds.
The class may be enhanced by parsing time period string like 00h, 00m, and 00s.

Thanks many helps,
Jongyeob

### CODE REVIEW

File: sunpy/net/jsoc/attrs.py [hunk 1]
```diff
@@ -3,17 +3,22 @@
...
+           'Segment','Sample']
 
 
 class Time(Time):
     """
     Time range to download
     """
-
+    
+class Sample(Sample):
```
Comment (Cadair) [inline sunpy/net/jsoc/attrs.py]: You don't need this here, just the import from vso will suffice, you will still be able to do `jsoc.attrs.Sample`

**Comment by Cadair** [comment]
thanks @jongyeob !!

This is looking great!

As for dealing with different units other than seconds, we should do that using the astropy quantity mechanism that has been adopted throughout the code base recently.

I think you would need to do a small but of input processing in `sunpy.net.vso.attrs.Sample` i.e:

```python

import astropy.units as u

class Sample(_VSOSimpleAttr):

    @u.quantity_input(value=u.s)
    def __init__(self, value):
        self.value = value.to(u.s).value
```

This would allow you do do things like:

```python
res = query(..., jsoc.Sample(12*u.hour)
```
to get a 12 hour sample rate.

**Comment by Cadair** [comment]
@jongyeob You also need to fix the test you changed (https://travis-ci.org/sunpy/sunpy/jobs/64948491#L1112)

It looks like you just need to add the `@90s` bit to the expected dataset in the payload string [here](https://github.com/sunpy/sunpy/blob/master/sunpy%2Fnet%2Fjsoc%2Ftests%2Ftest_jsoc.py#L47)

**Comment by jongyeob** [comment]
@Cadair , I followed your instructions. I applied a quantity decorator to validate only time quantities (e.g. ny, nw, nd, nh, nm, ns, etc. n : integer) on the sample class.

However, I don't know why travis CI failed.

File: sunpy/net/vso/attrs.py [hunk 2]
```diff
@@ -193,7 +193,12 @@ class Filter(_VSOSimpleAttr):
 
 
 class Sample(_VSOSimpleAttr):
-    pass
+    """
+    Time interval for data sampling
+    """
+    @u.quantity_input(value=u.s)
+    def __init__(self, value):
+        _VSOSimpleAttr.__init__(self,value)
```
Comment (Cadair) [inline sunpy/net/vso/attrs.py]: This should be written as 

```python
super(Sample,self).__init__(value)
```

**Comment by Cadair** [comment]
The Travis build is failing because of the documentation build, but I don't know why, I will try and take a look later.

**Comment by jongyeob** [comment]
I fixed small errors. Now the sample keyword is available.

**Comment by Cadair** [comment]
@jongyeob I have sent a PR to your fork to fix up this PR.

**Comment by jongyeob** [comment]
@Cadair , thank you for your kind descriptions. It was merged.

**Comment by Cadair** [comment]
:+1:

**Comment by Cadair** [comment]
@dpshelio @ayshih @sunpy/sunpy-maintainers Review please!

**Comment by ayshih** [comment]
It looks fine.  How about a changelog entry?

**Comment by Cadair** [comment]
@jongyeob Could you add a line to the `CHANGELOG.md` file describing the change please?

**Comment by jongyeob** [comment]
@Cadair Did I do well? AppVeyor is failed. I don't know why...

**Comment by Cadair** [comment]
@jongyeob can you merge upstream master into your branch?

**Comment by ayshih** [comment]
:+1:

**Comment by Cadair** [comment]
Thanks a lot @jongyeob that's great!

**Pull request merged**